File: src/main/java/com/decoder/AES.java
package com.decoder;

import com.decoder.Rijndael.AESMode;

public class AES {
	
//	private final String encoding = "UTF-8";
	public static byte[] KEY = "0123456789ABCDEF".getBytes();
	public static byte[] IV = "FEDCBA9876543210".getBytes();
	private Rijndael cbc_aes_pkcs5;
	
	public AES() {
		cbc_aes_pkcs5 = new Rijndael(KEY, IV, AESMode.CBC_PKCS5);
	}
	
	/**
	 * 解密
	 * @param by
	 * @return 
	 */
	public byte[] decrypt(byte[] by) {
		return cbc_aes_pkcs5.decryptHex(new String(by));
	}

	/**
	 * 加密
	 * @param bytes
	 * @return 
	 */
	public String encrypt(byte[] bytes) {
		return cbc_aes_pkcs5.encryptToHex(bytes);
	}
	
}

File: src/main/kotlin/com/decoder/AES.kt
package com.decoder

class AES {

    private val cbc_aes_pkcs5 = Rijndael(KEY, IV, AESMode.CBC_PKCS5)

    /**
     * 解密
     * @param by
     * @return
     */
    fun decrypt(by: ByteArray): ByteArray {
        return cbc_aes_pkcs5.decryptHex(by.decodeToString())
    }

    /**
     * 加密
     * @param bytes
     * @return
     */
    fun encrypt(bytes: ByteArray): String {
        return cbc_aes_pkcs5.encryptToHex(bytes)
    }

    companion object {
        //	private final String encoding = "UTF-8"
        var KEY: ByteArray = "0123456789ABCDEF".toByteArray()
        var IV: ByteArray = "FEDCBA9876543210".toByteArray()
    }

}
